pytorch_lightning/utilities/device_parser.py

```diff
@@ -107,13 +112,24 @@ def parse_tpu_cores(tpu_cores: Union[int, str, List]) -> Optional[Union[List[int
 
 
 def _normalize_parse_gpu_string_input(s: Union[int, str, List[int]]) -> Union[int, List[int]]:
-    if isinstance(s, str):
-        if s == '-1':
-            return -1
-        else:
-            return [int(x.strip()) for x in s.split(',') if len(x) > 0]
-    else:
+    if not isinstance(s, str):
         return s
+    if s == '-1':
+        return -1
+    elif ',' in s:
+        return [int(x.strip()) for x in s.split(',') if len(x) > 0]
+    else:
+        num_gpus = int(s.strip())
+        if _compare_version("pytorch_lightning", operator.lt, "1.5"):
+            rank_zero_warn(
+                f"Parsing of the Trainer argument gpus='{s}' (string) will change in the future."
+                " In the current version of Lightning, this will select"
```

Comment:
BTW Consider explicitly mentioning `int`; e.g. "but from v1.5 it will it will select gpus (same as gpus=d (int)) [...]"

Reply:
done, thanks!